File: problems-adapt/0001-0100/0063_majority-element/solution_sorting.cpp
class Solution {
  public:
    int majorityElement(vector<int> &nums) {
        // Sort in place: the majority's occurrences stand together as one
        // run longer than half the array, and a run that long must cover
        // the middle -- so the value at the halfway index is the majority,
        // whatever the input order was.
        sort(nums.begin(), nums.end());
        return nums[nums.size() / 2];
    }
};

File: problems-originals/0101-0200/0169_majority-element/solution_sorting.go
import "sort"

func majorityElement(nums []int) int {
	// Sort in place: the majority's occurrences stand together as one run
	// longer than half the array, and a run that long must cover the
	// middle -- so the value at the halfway index is the majority,
	// whatever the input order was.
	sort.Ints(nums)
	return nums[len(nums)/2]
}
